spec/controllers/votes_controller_spec.rb

```diff
@@ -1,4 +1,83 @@
 require 'rails_helper'
 
 RSpec.describe VotesController, type: :controller do
+    let(:user1) { create(:user_with_voteable_status) }
+    let(:user2) { create(:user_with_voteable_status) }
+    let(:user3) { create(:user_without_voteable_status) }
+    before do
+      create_list(:question, 5, user: user1)
+      create_list(:answer, 5, user: user2)
+    end
+
+  context "when user votes another user's resource" do
+    before do
+      request.headers['Authorization'] = "Token token=#{user2.refresh_token}"
+    end
+
+    it "upvotes another user's resource" do
+      post :upvote, {resource_name: 'questions', resource_id: 1}
+      expect(response.body).to eq "{\"response\":1}"
+      expect(response.status).to eq 200
+    end
+
+    it "downvotes another user's resource " do
+      post :downvote, {resource_name: 'questions', resource_id: 1}
+      expect(response.body).to eq "{\"response\":-1}"
+      expect(response.status).to eq 200
+    end
+  end
+
+  context "when user votes own resource" do
+    before do
+      request.headers['Authorization'] = "Token token=#{user2.refresh_token}"
+    end
+
+    it "doesnt downvote resource" do
+      post :downvote, {resource_name: 'answers', resource_id: 1}
+      expect(response.body).to include "You can't vote for your post"
+      expect(response.status).to eq 400
+    end
+
+    it "doesnt upvote resource" do
+      post :upvote, {resource_name: 'answers', resource_id: 1}
+      expect(response.body).to include "You can't vote for your post"
+      expect(response.status).to eq 400
+    end
+  end
+
+  context "when user cannot vote on resource" do
+    before do
+      request.headers['Authorization'] = "Token token=#{user3.refresh_token}"
+    end
+
+    it "doesnt upvotes another user's resource" do
+      post :upvote, {resource_name: 'questions', resource_id: 1}
+      expect(response.body).to include "Not qualified to vote"
+      expect(response.status).to eq 400
+    end
+
+    it "doesnt downvote another user's resource" do
+      post :downvote, {resource_name: 'answers', resource_id: 1}
+      expect(response.body).to include "Not qualified to vote"
+      expect(response.status).to eq 400
+    end
+  end
+
```

Comment:
I really can't tell where the error lies. My best suggestion would be the controller but this is supposed to return a status code of 403 not 400. Actually, all of these 400 statuses.